contracts/0.8.25/vaults/Permissions.sol

```diff
@@ -93,21 +100,44 @@ abstract contract Permissions is AccessControlVoteable {
         vaultHub = VaultHub(stakingVault().vaultHub());
         _grantRole(DEFAULT_ADMIN_ROLE, _defaultAdmin);
 
-        _setVoteLifetime(7 days);
+        _setConfirmLifetime(_confirmLifetime);
 
-        emit Initialized();
+        emit Initialized(_defaultAdmin);
     }
 
     function stakingVault() public view returns (IStakingVault) {
-        bytes memory args = Clones.fetchCloneArgs(address(this));
-        address addr;
-        assembly {
-            addr := mload(add(args, 32))
+        return IStakingVault(_loadStakingVaultAddress());
+    }
+
+    // ==================== Role Management Functions ====================
+
+    /**
+     * @notice Mass-grants multiple roles to multiple accounts.
+     * @param _assignments An array of role assignments.
```

Comment:
maybe note that duplicates wouldn't revert the call rather resulting in fewer events emitted (and cover this behavior with tests)